Services/Utilities/classes/class.ilUtil.php

```diff
@@ -189,23 +189,29 @@ public static function getStyleSheetLocation($mode = "output", $a_css_name = "",
 		}
 
 		$filename = "";
+		$ver_filename = "";
 		// use ilStyleDefinition instead of account to get the current skin
 		require_once("./Services/Style/classes/class.ilStyleDefinition.php");
 		if (ilStyleDefinition::getCurrentSkin() != "default")
 		{
 			$filename = "./Customizing/global/skin/".ilStyleDefinition::getCurrentSkin()."/".$a_css_location.$stylesheet_name;
+			$ver_filename = "./Customizing/global/skin/".ilStyleDefinition::getCurrentSkin()."/version";
 		}
 		if (strlen($filename) == 0 || !file_exists($filename))
 		{
 			$filename = "./" . $a_css_location . "templates/default/".$stylesheet_name;
 		}
 		$vers = "";
+        $build = "";
 		if ($mode != "filesystem")
 		{
 			$vers = str_replace(" ", "-", $ilias->getSetting("ilias_version"));
 			$vers = "?vers=".str_replace(".", "-", $vers);
+			if (strlen($ver_filename) != 0 && file_exists($ver_filename)) {
```

Comment:
file_exists is very expensive. Maybe really put that in the xml. We ready that anyways for the active skin/style.